src/arch/loongarch64.rs

```diff
@@ -0,0 +1,558 @@
+//! Low-level LoongArch64 support.
+//!
+//! This file is heavily based on the x86_64 implementation.
+//! Relevant differences are highlighted in comments, but otherwise most
+//! comments have been removed to avoid duplication. Refer to x86_64.rs for
+//! detailed comments about what is happening in this file.
+//!
+//! ## Stack layout
+//!
+//! Here is what the layout of the stack looks like when a coroutine is
+//! suspended.
+//!
+//! ```text
+//! +--------------+  <- Stack base
+//! | Initial func |
+//! +--------------+
+//! | Parent link  |
+//! +--------------+
+//! |              |
+//! ~     ...      ~
+//! |              |
+//! +--------------+
+//! | Padding      |
+//! +--------------+
+//! | Saved PC     |
+//! +--------------+
+//! | Saved FP     |
+//! +--------------+
+//! | Saved S0     |
```

Comment:
Just a reminder: all references to S0 need to be changed to S8. S0 can just be listed as a `lateout("$s0") _` instead of being saved/restored manually.

Reply:
Thanks for your reminder. :-)